nui-gestalt-di/build.gradle

```diff
@@ -0,0 +1,112 @@
+    inputs.files(sourceSets.main.output.classesDirs.files)
+    inputs.files(sourceSets.main.output.resourcesDir)
+
+    dependsOn(tasks.classes)
+    doLast {
+        def scanResultDir = sourceSets.main.output.resourcesDir
+
+        def inputs = inputs.files.files.toArray(new File[0])
+
+        new ClassGraph()
+                .overrideClasspath(inputs)
+                .enableClassInfo()
+                .acceptJars("nui-*.jar", "gestalt-*.jar")
+                .enableAnnotationInfo()
+                .verbose()
+                .enableRealtimeLogging()
+                .scan()
+                .withCloseable { scanResult ->
+                    def indexClasses = []
+                    indexClasses << scanResult.getClassesWithAnnotation("org.terasology.context.annotation.Index").directOnly()
+                    indexClasses << scanResult.getClassesWithAnnotation("org.terasology.context.annotation.IndexInherited").directOnly()
+                    indexClasses << indexingClasses.collect { scanResult.getClassInfo(it) }
+
+                    file("$scanResultDir/META-INF/subtypes/").mkdirs()
+                    file("$scanResultDir/META-INF/annotations/").mkdirs()
+
+                    indexClasses.flatten().each { ClassInfo classInfo ->
+                        if (classInfo.interface) {
+                            def file = file("$scanResultDir/META-INF/subtypes/$classInfo.name")
+
+                            file.newWriter().withCloseable { writer ->
+                                classInfo.classesImplementing.each {
+                                    writer.writeLine(it.name)
+                                }
+                            }
+                        } else if (classInfo.standardClass) {
+                            file("$scanResultDir/META-INF/subtypes/$classInfo.name")
+                                    .newWriter().withCloseable { writer ->
+                                classInfo.subclasses.each {
+                                    writer.writeLine(it.name)
+                                }
+                            }
+                        } else if (classInfo.annotation) {
+                            file("$scanResultDir/META-INF/annotations/$classInfo.name")
+                                    .newWriter().withCloseable { writer ->
+
+                                classInfo.classesWithAnnotation.each {
+                                    writer.writeLine(it.name)
+                                }
+                            }
+                        }
+                    }
+                    file("$scanResultDir/META-INF/resources")
+                            .newWriter().withCloseable { writer ->
+                        scanResult.allResources.filter {!it.path.startsWith("META-INF")}.nonClassFilesOnly().each { resource ->
+                            writer.writeLine(resource.getPath())
+                        }
+                    }
+                }
+    }
```

Comment:
These changes will not work as-is, as you did not generate a `META-INF/gestalt-indexes-present` file, which is used to identify which jars have been scanned by gestalt. It's a work-around for a very hard-to-diagnose android quirk. The file can be empty, as its contents is not read. Maybe something like this (untested)?
```groovy
file("$scanResultDir/META-INF/gestalt-indexes-present").createNewFile()
```